src/posix/main.c

```diff
@@ -293,10 +293,15 @@ static otInstance *InitInstance(PosixConfig *aConfig)
 
     syslog(LOG_INFO, "Running %s", otGetVersionString());
     syslog(LOG_INFO, "Thread version: %hu", otThreadGetVersion());
-    IgnoreError(otLoggingSetLevel(aConfig->mLogLevel));
 
     instance = otSysInit(&aConfig->mPlatformConfig);
     VerifyOrDie(instance != NULL, OT_EXIT_FAILURE);
+
+    // FIXME(wgtdkp): `otLoggingSetLevel` has the problem that
+    // it will (when dynamic logging is enabled) set value of
+    // the otInstance object but has no dependency on the
+    // creation of the otInstance object.
+    IgnoreError(otLoggingSetLevel(aConfig->mLogLevel));
```

Comment:
A simple fix would be making `otLogLevel mLogLevel;` static for `Instance` class?

Reply:
or move it into `src/common/logging.cpp` as a static local variable.

Reply:
Not sure if this is acceptable. Are we currently using only the `otInstance` singleton as static variable?
@jwhui @abtink

Reply:
Seems there are already examples of static member fields in `ot::Instance`, like `sHeap` and `sDnsNameCompressionEnabled`. Changed to set `mLoglevel` static in `ot::Instance`.

@bukepo I think it's easier to manage static variables in a single place (i.e. `ot::Instance`).

Reply:
I think putting it in logging.cpp can save a few code size. And I'm fine with it static in Instance.

Reply:
I agree making it a static member of `ot::Instance` makes sense. 
I think the fact that it is a member variable is left over from older code when the `otLog()` required `aInstance` to be passed as well.

> I think it's easier to manage static variables in a single place (i.e. ot::Instance).

Yes. Fully agree. It would be good to have all static data in one place (one class) so to be able to find and track them easier.